sdks/java/core/src/main/java/org/apache/beam/sdk/options/PipelineOptionsFactory.java

```diff
@@ -1792,4 +1804,82 @@ private synchronized void register(Class<? extends PipelineOptions> iface) {
       return combinedCache.get(interfaces).getPropertyDescriptors();
     }
   }
+
+  /**
+   * @param prefix a prefix filter on the map keys.
+   * @param options the options to convert to a pipeline options.
+   * @return a pipeline options instance based on the specified configuration.
+   */
+  @Experimental
+  public static PipelineOptions fromMap(final String prefix, final Map<String, String> options) {
+    checkArgument(prefix != null, "prefix should not be null.");
+    checkArgument(options != null, "options should not be null.");
+    final Map<String, String> filtered = options.entrySet()
+                                        .stream()
+                                        .filter(e -> e.getKey()
+                                                      .startsWith(prefix))
+                                        .collect(toMap(Map.Entry::getKey, Map.Entry::getValue));
+    return fromMap(filtered);
+  }
+
+  /**
+   * @param options the options to convert to a pipeline options.
+   * @return a pipeline options instance based on the specified configuration.
+   */
+  public static PipelineOptions fromMap(final Map<String, String> options) {
+    checkArgument(options != null, "options should not be null.");
+    return new Builder().fromMap(options).create();
+  }
+
+  /**
+   * @param properties the properties to use to create the pipeline options instance.
```

Comment:
Please improve the javadoc to be similar to `fromArgs` explaining the parsing.

It's also important to mention that you filter out non string key/value pairs since many people may assume that might call toString on everything.